mirrord/layer/src/lib.rs

```diff
@@ -606,21 +606,18 @@ pub(crate) unsafe extern "C" fn close_detour(fd: c_int) -> c_int {
         .get()
         .expect("Should be set during initialization!");
 
-    if SOCKETS.lock().unwrap().remove(&fd).is_some() {
-        FN_CLOSE(fd)
-    } else if *enabled_file_ops
+    let res = FN_CLOSE(fd);
+    if SOCKETS.lock().unwrap().remove(&fd).is_none() && *enabled_file_ops
         && let Some(remote_fd) = OPEN_FILES.lock().unwrap().remove(&fd) {
         let close_file_result = file::ops::close(remote_fd);
 
-        close_file_result
+        let _ = close_file_result
```

Comment:
Why shouldn't we pass the remote close result to the local process anymore?

Reply:
It's not an easy decision, best would be to return 0 if both are okay or return the invalid one, but what happens when both fail? Anyhow close is usually used as a best effort so I don't see too much value digging it